src/Razor/src/Microsoft.AspNetCore.Razor.Language/RazorParserFeatureFlags.cs

```diff
@@ -29,9 +36,14 @@ public static RazorParserFeatureFlags Create(RazorLanguageVersion version)
                 allowUsingVariableDeclarations = true;
             }
 
+            if (FileKinds.IsComponent(fileKind) || FileKinds.IsComponentImport(fileKind))
```

Comment:
`FileKinds.IsComponent` will return true for components imports as well. Can remove the other condition.